src/test/regress/expected/citus_local_tables_mx.out

```diff
@@ -773,6 +773,97 @@ SELECT logicalrelid, partmethod, partkey FROM pg_dist_partition
  parent_dropped_col_2 | h          | {VAR :varno 1 :varattno 5 :vartype 23 :vartypmod -1 :varcollid 0 :varlevelsup 0 :varnoold 1 :varoattno 5 :location -1}
 (2 rows)
 
+-- some tests for view propagation on citus local tables
+CREATE TABLE view_tbl_1 (a int);
+CREATE TABLE view_tbl_2 (a int);
+CREATE SCHEMA viewsc;
+-- create dependent views, in a different schema
+-- the first one depends on a citus metadata table
+CREATE VIEW viewsc.prop_view AS SELECT COUNT (*) FROM view_tbl_1 JOIN pg_dist_node ON view_tbl_1.a=pg_dist_node.nodeid;
+CREATE VIEW viewsc.prop_view2 AS SELECT COUNT (*) FROM view_tbl_1;
+SELECT citus_add_local_table_to_metadata('view_tbl_1');
```

Comment:
I think it is really valuable to have auto-converted citus local tables with views.

Partition auto convert:
```SQL
SET citus.use_citus_managed_tables TO ON;
CREATE TABLE parent_2 (a INT UNIQUE) PARTITION BY RANGE(a);
CREATE TABLE parent_1 (a INT UNIQUE) PARTITION BY RANGE(a);

SET citus.use_citus_managed_tables TO OFF;
CREATE TABLE parent_2_child_1 (a int);
CREATE TABLE parent_1_child_1 (a int);

CREATE VIEW v1 AS SELECT * FROM parent_2_child_1;
CREATE VIEW v2 AS SELECT * FROM parent_1_child_1;
CREATE VIEW v3 AS SELECT count(*) FROM v2 JOIN v1 ON (true);

alter table parent_1 attach partition parent_1_child_1 default ;
 alter table parent_2 attach partition parent_2_child_1 default ;

-- TODO: adjust schema names
select run_command_on_workers($$SELECT count(*) from sc101.v1, sc101.v2, sc101.v3$$);

 ```

fkey auto convert:
```SQL
create table ref(a int primary key);
SELECT create_reference_table('ref');

CREATE TABLE loc (a int );

CREATE VIEW v100 AS SELECT * FROM loc;
CREATE VIEW v101 AS SELECT * FROM loc JOIN ref USING (a);
CREATE VIEW v102 AS SELECT * FROM v101;

ALTER TABLE loc ADD CONSTRAINT fkey FOREIGN KEY (a) references ref(a);

-- works fine
select run_command_on_workers($$SELECT count(*) from sc101.v100, sc101.v101, sc101.v102$$);

--
ALTER TABLE loc DROP CONSTRAINT fkey;

-- fails because fkey is dropped and table is converted to local table
select run_command_on_workers($$SELECT count(*) from sc101.v100$$);
	select run_command_on_workers($$SELECT count(*) from sc101.v101$$);
	select run_command_on_workers($$SELECT count(*) from sc101.v102$$);
```